javalanglib/src/main/scala/java/lang/Enum.scala

```diff
@@ -0,0 +1,22 @@
+package java.lang
+
+abstract class Enum[E <: Enum[E]] protected(val name: String, val ordinal: Int)
+    extends Comparable[E] with java.io.Serializable {
+
+  override def toString(): String = name
+
+  override final def equals(that: Any): scala.Boolean = that match {
+    case that: Enum[E] if that.getClass == getClass =>
+      ordinal == that.ordinal && name == that.name
+
+    case _ => false
+  }
+
+  override final def hashCode(): Int =
+    31 * name.hashCode() + ordinal.hashCode()
+
+  final def compareTo(o: E): Int = ordinal.compareTo(o.ordinal)
```

Comment:
`protected final def clone()` is missing, between `hashCode` and `compareTo`. It should always throw `CloneNotSupportedException`.
http://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html#clone%28%29